test/framework/framework.go

```diff
@@ -577,16 +577,24 @@ func (f *Framework) DeletePrometheusOperatorClusterResource(ctx context.Context)
 }
 
 func (f *Framework) SetupPrometheusRBAC(ctx context.Context, t *testing.T, testCtx *TestCtx, ns string) {
-	if _, err := f.CreateOrUpdateClusterRole(ctx, fmt.Sprintf("%s/rbac/prometheus/prometheus-cluster-role.yaml", f.exampleDir)); err != nil && !apierrors.IsAlreadyExists(err) {
-		t.Fatalf("failed to create or update prometheus cluster role: %v", err)
-	}
 	if finalizerFn, err := f.createOrUpdateServiceAccount(ctx, ns, fmt.Sprintf("%s/rbac/prometheus/prometheus-service-account.yaml", f.exampleDir)); err != nil {
 		t.Fatal(errors.Wrap(err, "failed to create or update prometheus service account"))
 	} else {
 		if testCtx != nil {
 			testCtx.AddFinalizerFn(finalizerFn)
 		}
+	}
 
+	if _, err := f.CreateOrUpdateClusterRole(ctx, fmt.Sprintf("%s/rbac/prometheus/prometheus-cluster-role.yaml", f.exampleDir)); err != nil && !apierrors.IsAlreadyExists(err) {
+		t.Fatalf("failed to create or update prometheus cluster role: %v", err)
+	}
+
+	if finalizerFn, err := f.createOrUpdateClusterRoleBinding(ctx, ns, fmt.Sprintf("%s/rbac/prometheus/prometheus-cluster-role-binding.yaml", f.exampleDir)); err != nil {
```

Comment:
cluster role binding was required for `prometheus` so that `prometheus serviceAccount` has access to list Node objects otherwise prometheus errors following

```
ts=2023-07-03T05:47:22.289Z caller=klog.go:108 level=warn component=k8s_client_runtime func=Warningf msg="pkg/mod/k8s.io/client-go@v0.26.2/tools/cache/reflector.go:169: failed to list *v1.Node: nodes is forbidden: User \"system:serviceaccount:allns-y-scrapeconfigkubenode-rx7h5h-0-9252af28:prometheus\" cannot list resource \"nodes\" in API group \"\" at the cluster scope"
ts=2023-07-03T05:47:22.289Z caller=klog.go:116 level=error component=k8s_client_runtime func=ErrorDepth msg="pkg/mod/k8s.io/client-go@v0.26.2/tools/cache/reflector.go:169: Failed to watch *v1.Node: failed to list *v1.Node: nodes is forbidden: User \"system:serviceaccount:allns-y-scrapeconfigkubenode-rx7h5h-0-9252af28:prometheus\" cannot list resource \"nodes\" in API group \"\" at the cluster scope"
```

Reply:
I see that `SetupPrometheusRBACGlobal()` below configures the same rbac permissions. Can't we use it? I'm worried that changing the meaning of `SetupPrometheusRBAC()` hides problems in the long run (I wouldn't mind additional comments on the `Setup*RBAC*` functions to clarify what they do).